src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,193 @@
+    {
+        eventRaid(world, level, colony);
+    }
+
+    /**
+     * Commences a Raid event in a specific colony, with a specific RaidLevel
+     */
+
+    private static void eventRaid(final World raidingWorld, final int level, final Colony colony)
+    {
+        if (level == LEVEL_AT_WHICH_TO_NOT_TRIGGER_RAID)
+        {
+            return;
+        }
+
+        int numberOfBarbarians = level;
+        int numberOfArcherBarbarians = (int) (0.5 * level);
+        int numberOfChiefBarbarians = (int) (0.1 * level);
+
+        int hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+
+        if (hordeTotal > 40 && MAX_SIZE == 40 && hordeTotal > MAX_SIZE)
+        {
+            numberOfBarbarians = 22;
+            numberOfArcherBarbarians = 16;
+            numberOfChiefBarbarians = 2;
+        }
+
+        if (hordeTotal > Configurations.maxBarbarianHordeSize)
+        {
+
+            numberOfBarbarians = hordeTotal - MAX_SIZE;
+
+            //For error handling and correct hordeTotal
+            if (numberOfBarbarians < 0)
+            {
+                numberOfBarbarians = 0;
+            }
+
+            //Re-asses hordeTotal to see if it is now below or at MAX_SIZE
+            hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+
+            if (hordeTotal > MAX_SIZE)
+            {
+
+                numberOfArcherBarbarians = hordeTotal - MAX_SIZE;
+
+                //For error handling and correct hordeTotal
+                if (numberOfArcherBarbarians < 0)
+                {
+                    numberOfArcherBarbarians = 0;
+                }
+
+                //Re-asses hordeTotal to see if it is now below or at MAX_SIZE
+                hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+
+                if (hordeTotal > MAX_SIZE)
+                {
+
+                    numberOfChiefBarbarians = hordeTotal - MAX_SIZE;
+
+                    //For error handling and correct hordeTotal
+                    if (numberOfChiefBarbarians < 0)
+                    {
+                        numberOfChiefBarbarians = 0;
+                    }
+                }
+            }
+        }
+
+        //Make sure world isn't null
+        if (raidingWorld == null)
+        {
+            return;
+        }
+
+        int x = colony.getCenter().getX();
+        int y = colony.getCenter().getY();
+        int z = colony.getCenter().getZ();
+
+        switch (raidingWorld.rand.nextInt(7))
+        {
+            case 0:
+                x += Configurations.workingRangeTownHall + 20;
+                break;
+            case 1:
+                x -= Configurations.workingRangeTownHall + 20;
+                break;
+            case 2:
+                z += Configurations.workingRangeTownHall + 20;
+                break;
+            case 3:
+                z -= Configurations.workingRangeTownHall + 20;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 20 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)